### PR TITLE
[Urgent for Public Preview] Modify BackupShortTermRetentionPolicy Powershell Command

### PR DESCRIPTION
**Pull request by lululilliancoding**
## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by lululilliancoding** [comment]
How do we solve the problem that "Connect-AzAccount -SubscriptionId XXX" and "connect-azaccount -environment AzureCloud" will popup a window in black screen so blocked the login behavior?

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupShortTermRetentionPolicy.cs [hunk 1]
```diff
@@ -28,12 +29,19 @@ public class SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy : AzureSqlDatab
...
             Position = 3,
             HelpMessage = "The backup retention setting, in days.")]
-        [ValidateNotNullOrEmpty]
-        public int RetentionDays{ get; set; }
+        public int RetentionDays { get; set; }
 
+        /// <summary>
+        /// Gets or sets differential backup interval hours.
+        /// </summary>
+        [Parameter(Mandatory = false,
```
[the remark this answers is not in the document]
Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupShortTermRetentionPolicy.cs:40]: removed

File: src/Sql/Sql/help/Get-AzSqlDatabaseBackupShortTermRetentionPolicy.md [hunk 2]
```diff
@@ -32,17 +32,17 @@ Get-AzSqlDatabaseBackupShortTermRetentionPolicy -ResourceId <String> [-DefaultPr
...
 ```powershell
 PS C:\> Get-AzSqlDatabaseBackupShortTermRetentionPolicy -ResourceGroupName resourcegroup01 -ServerName server01 -DatabaseName database01
 
-ResourceGroupName ServerName  DatabaseName RetentionDays
------------------ ----------  ------------ -------------
-resourcegroup01   server01    database01   35
+ResourceGroupName ServerName  DatabaseName RetentionDays DiffBackupIntervalInHours
+----------------- ----------  ------------ ------------- -------------------------
+resourcegroup01   server01    database01   7             24
 ```
```
[the remark this answers is not in the document]
Comment (lululilliancoding) [reply src/Sql/Sql/help/Get-AzSqlDatabaseBackupShortTermRetentionPolicy.md:46]: regenerated and didn't touch content except examples and description

**Comment by ericshape** [comment]
@lululilliancoding there are conflicts in the csproj files. please rebase and address them

File: src/Sql/Sql/Sql.csproj [hunk 3]
```diff
@@ -21,7 +21,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Management.Sql" Version="1.53.0-preview" /> 
+    <PackageReference Include="Microsoft.Azure.Management.Sql" Version="1.55.0-preview" />
```
[the remark this answers is not in the document]
Comment (lululilliancoding) [reply src/Sql/Sql/Sql.csproj:24]: 1.55.0 now released to nuget.

File: src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupShortTermRetentionPolicy.cs [hunk 4]
```diff
@@ -28,12 +29,19 @@ public class SetAzureRmSqlDatabaseBackupShortTermRetentionPolicy : AzureSqlDatab
...
+            Position = 4,
+            HelpMessage = "The differential backup interval, in hours.")]
+        public int DiffBackupIntervalInHours { get; set; }
```
[the remark this answers is not in the document]
Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Cmdlet/SetAzureSqlDatabaseBackupShortTermRetentionPolicy.cs:43]: synced offline, this scenario doesn't included.

File: src/Sql/Sql/Database Backup/Model/AzureSqlDatabaseBackupShortTermRetentionPolicyModel.cs [hunk 5]
```diff
@@ -34,7 +36,12 @@ public class AzureSqlDatabaseBackupShortTermRetentionPolicyModel
         /// <summary>
         /// Gets or sets the retention days of this policy.
         /// </summary>
-        public int RetentionDays { get; set; }
+        public int? RetentionDays { get; set; }
```
[the remark this answers is not in the document]
Comment (lululilliancoding) [reply src/Sql/Sql/Database Backup/Model/AzureSqlDatabaseBackupShortTermRetentionPolicyModel.cs:39]: synced offline, this is not breaking change.

**Review by ericshape — APPROVED** [review]

File: src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md [hunk 6]
```diff
@@ -14,49 +14,59 @@ Sets a backup short term retention policy.
...
- [-Confirm] [<CommonParameters>]
+Set-AzSqlDatabaseBackupShortTermRetentionPolicy [-RetentionDays <Int32>] [-DiffBackupIntervalInHour <Int32>]
+ [-ResourceGroupName] <String> [-ServerName] <String> [-DatabaseName] <String>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
 ```
 
 ### PolicyByInputObjectSet
 ```
-Set-AzSqlDatabaseBackupShortTermRetentionPolicy [-RetentionDays] <Int32>
+Set-AzSqlDatabaseBackupShortTermRetentionPolicy [-RetentionDays <Int32>] [-DiffBackupIntervalInHour <Int32>]
```
Comment (BethanyZhou) [inline src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md:24]: Please update help files. DiffBackupIntervalInHour should be DiffBackupIntervalInHours

File: src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md [hunk 7]
```diff
@@ -14,49 +14,59 @@ Sets a backup short term retention policy.
 
 ### PolicyByResourceServerDatabaseSet (Default)
 ```
-Set-AzSqlDatabaseBackupShortTermRetentionPolicy [-RetentionDays] <Int32> [-ResourceGroupName] <String>
- [-ServerName] <String> [-DatabaseName] <String> [-DefaultProfile <IAzureContextContainer>] [-WhatIf]
- [-Confirm] [<CommonParameters>]
+Set-AzSqlDatabaseBackupShortTermRetentionPolicy [-RetentionDays <Int32>] [-DiffBackupIntervalInHour <Int32>]
```
Comment (BethanyZhou) [inline src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md:17]: DiffBackupIntervalInHour ->DiffBackupIntervalInHours

**Review by BethanyZhou — COMMENTED** [review]
Please add a change msg for your change

File: src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md (hunk 7 above)
Comment (lululilliancoding) [reply src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md]: modified

File: src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md (hunk 6 above)
Comment (lululilliancoding) [reply src/Sql/Sql/help/Set-AzSqlDatabaseBackupShortTermRetentionPolicy.md]: modified

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**